analyzer/aggregate_stats/aggregate_stats.go

```diff
@@ -0,0 +1,282 @@
+			layer:               layer,
+			dbTable:             "stats.daily_active_accounts",
+			dbColumn:            "active_accounts",
+			consensusStatsQuery: queries.ConsensusActiveAccounts,
+			runtimeStatsQuery:   queries.RuntimeActiveAccounts,
+			windowSize:          24 * time.Hour,
+			windowStep:          5 * time.Minute,
+			target:              a.target,
+		})
+	}
+
+	for {
+		for _, statsComputation := range statsComputations {
+			logger := a.logger.With("name", statsComputation.Name(), "layer", statsComputation.Layer())
+
+			windowSize := statsComputation.WindowSize()
+			windowStep := statsComputation.WindowStep()
+			// Rounds the timestamp down to the nearest windowStep interval.
+			// e.g. if windowStep is 5 minutes, then 12:34:56 will be rounded down to 12:30:00.
+			floorWindow := func(ts *time.Time) time.Time {
+				return ts.Truncate(windowStep)
+			}
+
+			logger.Info("updating stats")
+
+			// First we find the start and end timestamps for which we can compute stats.
+			// Start at the latest already computed window, or at the earliest indexed
+			// block if no stats have been computed yet.
+			latestComputed, err := statsComputation.LatestComputedTs(ctx)
+			switch {
+			case err == nil:
+				// Continues below.
+			case errors.Is(pgx.ErrNoRows, err):
+				// No stats yet. Start at the earliest indexed block.
+				var earliestBlockTs *time.Time
+				earliestBlockTs, err = a.earliestBlockTs(ctx, statsComputation.Layer())
+				if err != nil {
+					logger.Error("failed querying earliest indexed block timestamp", "err", err)
+					continue
+				}
+				latestComputed = floorWindow(earliestBlockTs)
+			default:
+				logger.Error("failed querying latest computed stats window", "err", err)
+				continue
+			}
+
+			// End at the latest indexed block timestamp rounded down to the windowStep interval.
+			// Unless a customLatestAvailableTs function is provided, in which case use that.
+			var latestPossibleWindow time.Time
+			if fn := statsComputation.customLatestAvailableTs; fn == nil {
+				// Use the latest indexed block timestamp for the layer.
+				latestBlockTs, err := a.latestBlockTs(ctx, statsComputation.Layer())
+				if err != nil {
+					// Log this as info as this can be expected when there's no indexed blocks yet, or if any of the layers are not being indexed.
+					logger.Info("failed querying latest indexed block timestamp, skipping iteration", "err", err)
+					continue
+				}
+				latestPossibleWindow = floorWindow(latestBlockTs)
+			} else {
+				// Compute the latest available window using the provided custom function.
+				ts, err := fn(ctx, a.target)
+				if err != nil {
+					// Log this as info as this can be expected when there's no indexed blocks yet, or if any of the layers are not being indexed.
+					logger.Info("failed querying the latest possible window, skipping iteration", "err", err)
+					continue
+				}
+				latestPossibleWindow = floorWindow(ts)
```

Comment:
Look like we can dedupe this block with its equivalent from `fn == nil`? 

Optionally, we could simplify away the whole `if ... customLatestAvailableTs == nil` if we forced every `statsComputation` to specify a fn. It does introduce a little more abstraction though, so your call on what you prefer.

Reply:
> It does introduce a little more abstraction though, so your call on what you prefer.

Maybe makes sense, yeah, will see how it looks.

Reply:
Updated this as suggested. Also, after taking another look at this PR after some time has passed, I did some other (pretty minor) code updates that I think increase the readability a bit.

Reply:
Thank you! Since there is only one squashed and rebased commit, I don't think there's a good way for me to see the diff, so I'll take your word for it :)